frontend/src/metabase/dashboards/containers/Dashboards.jsx

```diff
@@ -72,35 +108,68 @@ export class Dashboards extends Component {
         );
     }
 
+    searchTextFilter = (searchText: string) =>
+        ({name, description}: Dashboard) =>
+            (caseInsensitiveSearch(name, searchText) || (description && caseInsensitiveSearch(description, searchText)))
+
+    sectionFilter = (section: ListFilterWidgetItem) =>
+        ({creator_id, favorite}: Dashboard) =>
+        (section.id === SECTION_ID_ALL) ||
+        (section.id === SECTION_ID_MINE && creator_id === this.props.user.id) ||
+        (section.id === SECTION_ID_FAVORITES && favorite === true)
+
     getFilteredDashboards = () => {
-        const {searchText} = this.state;
+        const {searchText, section} = this.state;
         const {dashboards} = this.props;
+        const noOpFilter = _.constant(true)
 
-        if (searchText === "") {
-            return dashboards;
-        } else {
-            return dashboards.filter(({name, description}) =>
-                caseInsensitiveSearch(name,searchText) || (description && caseInsensitiveSearch(description, searchText))
-            );
-        }
+        return _.chain(dashboards)
+            .filter(searchText != "" ? this.searchTextFilter(searchText) : noOpFilter)
+            .filter(this.sectionFilter(section))
+            .value()
+            .sort((a, b) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()))
+    }
+
+    updateSection = (section: ListFilterWidgetItem) => {
+        this.setState({section});
     }
 
     render() {
-        let {modalOpen, searchText} = this.state;
+        let {modalOpen, searchText, section} = this.state;
 
         const isLoading = this.props.dashboards === null
         const noDashboardsCreated = this.props.dashboards && this.props.dashboards.length === 0
         const filteredDashboards = isLoading ? [] : this.getFilteredDashboards();
-        const noSearchResults = searchText !== "" && filteredDashboards.length === 0;
+        const noResultsFound = filteredDashboards.length === 0;
 
         return (
             <LoadingAndErrorWrapper
                 loading={isLoading}
-                className={cx("relative mx4", {"flex-full flex align-center justify-center": noDashboardsCreated})}
+                className={cx("relative mx4", {"flex flex-full flex-column": noDashboardsCreated})}
             >
                 { modalOpen ? this.renderCreateDashboardModal() : null }
+                <div className="flex align-center pt4 pb1">
+                    <TitleAndDescription title="Dashboards"/>
+
+                    <div className="flex-align-right cursor-pointer text-grey-5">
+                        <Link to="/dashboards/archive">
+                            <Icon name="viewArchive"
+                                  className="mr2 text-brand-hover Icon Icon-viewArchive"
```

Comment:
`Icon` class  doesn't currently merge them to the provided `className` parameter but I will make it do that :-)